File: CS/GrowthWare.Framework/Model/Profiles/Base/MProfile.cs
using GrowthWare.Framework.Model.Profiles.Interfaces;
using System;
using System.Collections.Generic;
using System.Data;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GrowthWare.Framework.Model.Profiles.Base
{
    /// <summary>
    /// Base class for profile objects.
    /// </summary>
    [Serializable()]
    public abstract class MProfile : MDatabaseFunctions, IMProfile
    {

        int m_Id = -1;

        #region Public Properties
        /// <summary>
        /// Account ID used to add
        /// </summary>
        public int AddedBy { get; set; }

        /// <summary>
        /// Date the row was added.
        /// </summary>
        public DateTime AddedDate { get; set; }

        public int Id
        {
            get { return m_Id; }
            set { m_Id = value; }
        }

        public string IdColumnName { get; set; }

        public string Name { get; set; }

        public string NameColumnName { get; set; }

        public int UpdatedBy { get; set; }

        public DateTime UpdatedDate { get; set; }
        #endregion

        #region Protected Methods
        /// <summary>
        /// Initializes values given a DataRow
        /// </summary>
        /// <param name="dataRow">datarow</param>
        /// <remarks>
        /// Only sets Id and Name if IdColumnName or NameColumnName is not null
        /// </remarks>
        internal virtual void Initialize(DataRow dataRow)
        {
            this.AddedBy = base.GetInt(dataRow, "Added_By");
            this.AddedDate = base.GetDateTime(dataRow, "Added_Date", DateTime.Now);
            this.UpdatedBy = base.GetInt(dataRow, "Updated_By");
            this.UpdatedDate = base.GetDateTime(dataRow, "Updated_Date", DateTime.Now);
            if (!String.IsNullOrEmpty(this.NameColumnName)) 
            {
                this.Name = base.GetString(dataRow, this.NameColumnName);
            }
            if (!String.IsNullOrEmpty(this.IdColumnName)) 
            {
                this.Id = base.GetInt(dataRow, this.IdColumnName);
            }
        }

        #endregion
    }
}

File: VB/GrowthWare.Framework/Model/Profiles/Base/MProfile.vb
Imports GrowthWare.Framework.Model.Profiles.Interfaces
Namespace Model.Profiles.Base
    ''' <summary>
    ''' Contains all profile base properties
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable(), CLSCompliant(True)> _
    Public MustInherit Class MProfile
        Inherits MDatabaseFunctions
        Implements IMProfile

        Public Property AddedBy As Integer Implements IMProfile.AddedBy

        Public Property AddedDate As Date Implements IMProfile.AddedDate

        Public Property Id As Integer = -1 Implements IMProfile.Id

        Public Property IdColumnName As String Implements IMProfile.IdColumnName

        Public Property Name As String Implements IMProfile.Name

        Public Property NameColumnName As String Implements IMProfile.NameColumnName

        Public Property UpdatedBy As Integer Implements IMProfile.UpdatedBy

        Public Property UpdatedDate As Date Implements IMProfile.UpdatedDate

        ''' <summary>
        ''' Initializes values given a DataRow
        ''' </summary>
        ''' <param name="Datarow">datarow</param>
        ''' <remarks>Only sets Id and Name if IdColumnName or NameColumnName is not null</remarks>
        Friend Overridable Sub Initialize(ByVal dataRow As DataRow)
            Me.AddedBy = Me.GetInt(dataRow, "Added_By")
            Me.AddedDate = Me.GetDateTime(dataRow, "Added_Date", DateTime.Now)
            Me.UpdatedBy = Me.GetInt(dataRow, "Updated_By")
            Me.UpdatedDate = Me.GetDateTime(dataRow, "Updated_Date", DateTime.Now)
            If Not String.IsNullOrEmpty(Me.IdColumnName) Then
                Me.Id = Me.GetInt(dataRow, Me.IdColumnName)
            End If
            If Not String.IsNullOrEmpty(Me.NameColumnName) Then
                Me.Name = Me.GetString(dataRow, Me.NameColumnName)
            End If
        End Sub
    End Class
End Namespace
